render_test.go

```diff
@@ -174,6 +175,217 @@ func TestLinesToTracebackRender(t *testing.T) {
 	}
 }
 
+func TestDiagnosticsOnArrowKeys(t *testing.T) {
+	const selectAFrom = "select a from"
+	scenarios := []struct {
+		previousText     string
+		nextText         string
+		linesToTraceBack int
+		lastKey          Key
+	}{
+		{previousText: selectAFrom, nextText: selectAFrom, linesToTraceBack: 0, lastKey: Up},
+		{previousText: selectAFrom, nextText: selectAFrom, linesToTraceBack: 0, lastKey: Left},
+		{previousText: selectAFrom, nextText: selectAFrom, linesToTraceBack: 0, lastKey: Right},
+		{previousText: selectAFrom, nextText: selectAFrom, linesToTraceBack: 0, lastKey: Down},
+		{previousText: selectAFrom, nextText: selectAFrom, linesToTraceBack: 0, lastKey: Escape},
+	}
+
+	buf := make([]byte, 1024)
+	writer := VT100Writer{buffer: buf}
+	posixWriter := &PosixWriter{writer, 0}
+	diagnostics := []lsp.Diagnostic{{
+		Range: lsp.Range{
+			Start: lsp.Position{Line: 0, Character: 0},
+			End:   lsp.Position{Line: 0, Character: 10},
+		},
+		Message: "mock message",
+	}}
+
+	r := &Render{
+		diagnostics:        diagnostics,
+		out:                posixWriter,
+		livePrefixCallback: func() (string, bool) { return "", false },
+		col:                100,
+		row:                100,
+	}
+
+	for idx, s := range scenarios {
```

Comment:
```suggestion
	for idx, scenario := range scenarios {
```

Reply:
Hmmm, we have this pattern in multiple tests and I think I'd rather keep s since we repeat it really often across the tests. Would that be ok?